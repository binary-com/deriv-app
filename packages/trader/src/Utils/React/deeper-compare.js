/**
 * Deep compare component's props and state for re-render check.
 *
 * @return {Boolean} Whether we should re-render the component or not
 */
export const deeperCompare = (next_props, next_state, this_props, this_state) => {
    let has_prop_changed = false;
    let has_state_changed = false;

    if (isComparable(next_props)) {
        has_prop_changed = Object.keys(next_props)
            .some(prop => typeof next_props[prop] !== 'function' &&
            typeof next_props[prop] === 'object' ?
                JSON.stringify(next_props[prop]) !== JSON.stringify(this_props[prop]) :
                next_props[prop] !== this_props[prop]);
    }

    if (isComparable(next_state)) {
        has_state_changed = Object.keys(next_state).some(state => next_state[state] !== this_state[state]);
    }

    return has_prop_changed || has_state_changed;
};

const isComparable = (value) => !!value &&  typeof value === 'object';
