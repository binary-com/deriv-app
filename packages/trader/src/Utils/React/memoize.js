export function memoize(func) {
    const cache = {};
    return function (...args) {
        const key = JSON.stringify(args);
        if (key in cache) return cache[key];
        return (cache[key] = func(...args));
    };
}
