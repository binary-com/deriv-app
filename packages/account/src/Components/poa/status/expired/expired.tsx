import { Button, Icon, Text } from '@deriv/components';
import { Localize, localize } from '@deriv/translations';
import React from 'react';
import IconMessageContent from '../../../icon-message-content';

type TExpired = {
    onClick: () => void;
};

export const Expired = ({ onClick }: TExpired) => (
    <div className='account-management__container'>
        <IconMessageContent
            message={localize('New proof of address is needed')}
            text={
                <Localize
                    i18n_default_text='Your document for proof of address is expired. <0/>Please submit again.'
                    components={[<br key={0} />]}
                />
            }
            icon={<Icon icon='IcPoaUpload' size={128} />}
        >
            <Button onClick={onClick} has_effect primary>
                <Text className='dc-btn__text' size='xs' weight='bold' as='p'>
                    {localize('Resubmit')}
                </Text>
            </Button>
        </IconMessageContent>
    </div>
);
