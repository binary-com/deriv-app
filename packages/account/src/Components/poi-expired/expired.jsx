import React from 'react';
import { PropTypes } from 'prop-types';
import { Icon } from '@deriv/components';
import { Localize, localize } from '@deriv/translations';
import IconMessageContent from 'Components/icon-message-content';

const Expired = ({ is_dashboard }) => (
    <IconMessageContent
        message={localize('New proof of identity document needed')}
        text={
            <Localize
                i18n_default_text='Kindly send a scan of a valid proof of identity to <0>support@deriv.com</0>'
                components={[
                    <a
                        key={0}
                        className='link link--orange'
                        rel='noopener noreferrer'
                        target='_blank'
                        href='mailto:support@deriv.com'
                    />,
                ]}
            />
        }
        icon={
            is_dashboard ? (
                <Icon icon='IcPoiUploadDashboard' height={128} width={237} />
            ) : (
                <Icon icon='IcPoiUpload' size={128} />
            )
        }
        className='account-management-dashboard'
    />
);

Expired.propTypes = {
    is_dashboard: PropTypes.bool,
};

export default Expired;
