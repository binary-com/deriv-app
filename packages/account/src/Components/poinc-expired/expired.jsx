import { Button, Icon, Text } from '@deriv/components';
import { localize } from '@deriv/translations';
import React from 'react';
import IconMessageContent from 'Components/icon-message-content';

export const Expired = ({ onClick }) => (
    // need to be refactored

    <IconMessageContent
        message={localize('New proof of income is needed')}
        text={localize('Your documents for proof of income is expired. Please submit again.')}
        icon={<Icon icon='IcPoaUpload' size={128} />}
    >
        <Button onClick={onClick} has_effect primary>
            <Text className='dc-btn__text' size='xs' weight='bold' as='p'>
                {localize('Resubmit')}
            </Text>
        </Button>
    </IconMessageContent>
);
