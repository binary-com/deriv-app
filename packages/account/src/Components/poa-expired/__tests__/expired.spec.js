import React from 'react';
import { screen, render } from '@testing-library/react';
import { Expired } from '../expired';

describe('<Expired/>', () => {
    const message = 'New proof of address is needed';
    const text = 'Your documents for proof of address is expired. Please submit again.';
    const onClick = jest.fn();

    it('should render Expired component', () => {
        render(<Expired onClick={onClick} />);

        expect(screen.getByText(message)).toBeInTheDocument();
        expect(screen.getByText(text)).toBeInTheDocument();
        expect(screen.getByText('Resubmit')).toBeInTheDocument();
        expect(screen.getByRole('button')).toBeInTheDocument();
    });

    it('should render Expired component snapshot check', () => {
        const tree = render(<Expired onClick={onClick} />);

        expect(tree).toMatchSnapshot();
    });
});
