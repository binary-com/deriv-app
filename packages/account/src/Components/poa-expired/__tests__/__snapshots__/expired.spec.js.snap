// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Expired/> should render Expired component snapshot check 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="account-management__message-wrapper"
      >
        <div
          class="account-management__message-content"
        >
          <div
            class="account-management__message-icon"
          >
            <svg
              class="dc-icon"
              height="128"
              width="128"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              <use
                xlink:href="/public/sprites/common.784d255e58485e5ee9f70cc9a40c7ffe.svg#ic-poa-upload"
              />
            </svg>
          </div>
          <div
            class="dc-text account-management__message"
            style="--text-size: var(--text-size-s); --text-color: var(--text-general); --text-lh: var(--text-lh-m); --text-weight: var(--text-weight-bold); --text-align: var(--text-align-center);"
          >
            New proof of address is needed
          </div>
          <div
            class="account-management__text-container"
          >
            <p
              class="dc-text"
              style="--text-size: var(--text-size-xs); --text-color: var(--text-general); --text-lh: var(--text-lh-m); --text-weight: var(--text-weight-normal); --text-align: var(--text-align-center);"
            >
              Your documents for proof of address is expired. Please submit again.
            </p>
          </div>
          <button
            class="dc-btn dc-btn__effect dc-btn--primary"
            tabindex="0"
            type="submit"
          >
            <span
              class="dc-text dc-btn__text"
              style="--text-size: var(--text-size-xs); --text-color: var(--text-general); --text-lh: var(--text-lh-m); --text-weight: var(--text-weight-bold); --text-align: var(--text-align-center);"
            >
              <p
                class="dc-text dc-btn__text"
                style="--text-size: var(--text-size-xs); --text-color: var(--text-general); --text-lh: var(--text-lh-m); --text-weight: var(--text-weight-bold); --text-align: var(--text-align-left);"
              >
                Resubmit
              </p>
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="account-management__message-wrapper"
    >
      <div
        class="account-management__message-content"
      >
        <div
          class="account-management__message-icon"
        >
          <svg
            class="dc-icon"
            height="128"
            width="128"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <use
              xlink:href="/public/sprites/common.784d255e58485e5ee9f70cc9a40c7ffe.svg#ic-poa-upload"
            />
          </svg>
        </div>
        <div
          class="dc-text account-management__message"
          style="--text-size: var(--text-size-s); --text-color: var(--text-general); --text-lh: var(--text-lh-m); --text-weight: var(--text-weight-bold); --text-align: var(--text-align-center);"
        >
          New proof of address is needed
        </div>
        <div
          class="account-management__text-container"
        >
          <p
            class="dc-text"
            style="--text-size: var(--text-size-xs); --text-color: var(--text-general); --text-lh: var(--text-lh-m); --text-weight: var(--text-weight-normal); --text-align: var(--text-align-center);"
          >
            Your documents for proof of address is expired. Please submit again.
          </p>
        </div>
        <button
          class="dc-btn dc-btn__effect dc-btn--primary"
          tabindex="0"
          type="submit"
        >
          <span
            class="dc-text dc-btn__text"
            style="--text-size: var(--text-size-xs); --text-color: var(--text-general); --text-lh: var(--text-lh-m); --text-weight: var(--text-weight-bold); --text-align: var(--text-align-center);"
          >
            <p
              class="dc-text dc-btn__text"
              style="--text-size: var(--text-size-xs); --text-color: var(--text-general); --text-lh: var(--text-lh-m); --text-weight: var(--text-weight-bold); --text-align: var(--text-align-left);"
            >
              Resubmit
            </p>
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
